### PR TITLE
US128797: Remove legacy version of My Courses Widget.

### PR DESCRIPTION
**Pull request by njostonehouse**

### CODE REVIEW

File: test/index.html [hunk 1]
```diff
@@ -16,7 +16,7 @@
 				'./d2l-my-courses-filter/d2l-my-courses-filter.html',
 				'./d2l-my-courses-search/d2l-my-courses-search.html',
 				'./d2l-utility-helpers/d2l-utility-helpers.html',
-				'./localize-behavior/localize-behavior.html'
+				'./localize-behavior/localize-behavior.html',
```
Comment (njostonehouse) [inline test/index.html:19]: oops

File: .gitignore [hunk 2]
```diff
@@ -4,5 +4,4 @@ node_modules*/
 node_modules
 package-lock.json
 
-test/index.html
```
Comment (njostonehouse) [inline .gitignore:7]: Any idea why this might have been ignored?

Comment (svanherk) [reply .gitignore:7]: I think back when we were using Travis for CI, we created an `index.html` file out of either `index.all.html` or `index.high-priority.html` depending on whether it was a pull request or a merge run.  And then Travis got mad if we tried to publish with uncommitted changes.  Should be good to stop ignoring now!

File: d2l-my-courses.js [hunk 3]
```diff
@@ -41,61 +36,32 @@ class MyCourses extends PolymerElement {
 			promotedSearches: String,
```
Comment (svanherk) [inline d2l-my-courses.js:36]: Above, I think we can also remove `showCourseCode`, `courseUpdatesConfig`, and `showSemester` that were only passed into `d2l-my-courses-legacy`.  Not sure if these got cleaned up in the lms or if we can remove the passing in there too

Comment (njostonehouse) [reply d2l-my-courses.js:36]: I will check in the LMS and see. If so, I will clean these up in a future PR.

File: d2l-my-courses.js [hunk 4]
```diff
@@ -41,61 +36,32 @@ class MyCourses extends PolymerElement {
...
+				promoted-searches="[[promotedSearches]]"
+				user-settings-url="[[userSettingsUrl]]"
+				org-unit-type-ids="[[orgUnitTypeIds]]"
+				course-image-upload-cb="[[courseImageUploadCb]]"
+				token="[[token]]">
+			</d2l-my-courses-container>`;
 	}
 
 	courseImageUploadCompleted(success) {
-		return this.updatedSortLogic
```
Comment (svanherk) [inline d2l-my-courses.js:91]: Do these functions still need to `return`?

Comment (njostonehouse) [reply d2l-my-courses.js:91]: I imagine so - my mistake.

File: package.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
   "name": "d2l-my-courses",
   "scripts": {
     "lint": "npm run lint:wc && npm run lint:js",
-    "lint:js": "eslint . --ext .js,.html src/**/*.js test/**/*.js test/**/*.html demo/**/*.html legacy/**/*.js legacy-test/**/*.js legacy-test/**/*.html",
+    "lint:js": "eslint . --ext .js,.html src/**/*.js test/**/*.js test/**/*.html demo/**/*.html",
     "lint:wc": "polymer lint",
     "test": "npm run lint && polymer test --config-file wct.conf.local.json"
   },
```
Comment (svanherk) [inline package.json:19]: We should now be able to cleanup a bunch of dependencies that were only used in legacy:
- `@polymer/iron-a11y-announcer` (only used `/legacy/tile-grid/d2l-course-tile.js`)
- `@polymer/iron-pages` (only used in `/legacy/search-filter/d2l-filter-menu-legacy.js`)
- `@polymer/iron-scroll-threshold` (only used in `/legacy/d2l-all-courses-legacy.js` and `/legacy/tile-grid/d2l-my-courses-content-animated.js`)
- `d2l-course-image` (only used in `/legacy/tile-grid/d2l-course-tile.js`)
- `d2l-menu` (used in many places, but only legacy)
- `d2l-organization-hm-behavior` (only used in `/legacy/d2l-all-courses-legacy.js` and `/legacy/tile-grid/d2l-course-tile.js`)
- `d2l-polymer-siren-behaviors` (only used in `/legacy/d2l-my-courses-content-behavior-legacy.js`)
- `d2l-search-widget` (used in many places, but only legacy)
- `d2l-simple-overlay` (used in many places, but only legacy)

Been waiting a long time for this! 🎉

Comment (njostonehouse) [reply package.json:19]: Oh boy!!

**Review by svanherk — APPROVED** [review]
Woohoo!!! 🎉 🎉 🎉

**Pull request merged**